velox/functions/sparksql/String.h

```diff
@@ -859,6 +865,9 @@ struct TranslateFunction {
     int k = 0;
     while (k < input.size()) {
       int inputCharLength = utf8proc_char_length(input.data() + k);
+      if (k + inputCharLength > input.size()) {
+        inputCharLength = input.size() - k;
+      }
```

Comment:
Thanks for the fix. Does it make sense to extract a private function to get the length?

Reply:
Updated, thanks.